tests/e2e/pipeline.go

```diff
@@ -333,3 +338,140 @@ func startPipelineRolloutWatches() {
+
+	pipelineRolloutSpec := createPipelineRolloutSpec(name, namespace, spec)
+	_, err := pipelineRolloutClient.Create(ctx, pipelineRolloutSpec, metav1.CreateOptions{})
+	Expect(err).ShouldNot(HaveOccurred())
+
+	document("Verifying that the PipelineRollout was created")
+	Eventually(func() error {
+		_, err := pipelineRolloutClient.Get(ctx, name, metav1.GetOptions{})
+		return err
+	}, testTimeout, testPollingInterval).Should(Succeed())
+
+	document("Verifying that the Pipeline was created")
+	verifyPipelineSpec(namespace, name, func(retrievedPipelineSpec numaflowv1.PipelineSpec) bool {
+		return len(spec.Vertices) == len(retrievedPipelineSpec.Vertices) // TODO: make less kludgey
+		//return reflect.DeepEqual(pipelineSpec, retrievedPipelineSpec) // this may have had some false negatives due to "lifecycle" field maybe, or null values in one
+	})
+
+	verifyPipelineRolloutDeployed(name)
+	verifyPipelineRolloutHealthy(name)
+	verifyInProgressStrategy(name, apiv1.UpgradeStrategyNoOp)
+
+	verifyPipelineRunning(namespace, name)
+
+}
+
+func createPipelineRolloutSpec(name, namespace string, pipelineSpec numaflowv1.PipelineSpec) *apiv1.PipelineRollout {
+
+	pipelineSpecRaw, err := json.Marshal(pipelineSpec)
+	Expect(err).ShouldNot(HaveOccurred())
+
+	pipelineRollout := &apiv1.PipelineRollout{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "numaplane.numaproj.io/v1alpha1",
+			Kind:       "PipelineRollout",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      name,
+		},
+		Spec: apiv1.PipelineRolloutSpec{
+			Pipeline: apiv1.Pipeline{
+				Spec: runtime.RawExtension{
+					Raw: pipelineSpecRaw,
+				},
+			},
+		},
+	}
+
+	return pipelineRollout
+
+}
+
+// delete a PipelineRollout and verify deletion
+func deletePipelineRollout(name string) {
+
+	document("Deleting PipelineRollout")
+	err := pipelineRolloutClient.Delete(ctx, name, metav1.DeleteOptions{})
+	Expect(err).ShouldNot(HaveOccurred())
+
+	document("Verifying PipelineRollout deletion")
+	Eventually(func() bool {
+		_, err := pipelineRolloutClient.Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				Fail("An unexpected error occurred when fetching the PipelineRollout: " + err.Error())
+			}
+			return false
+		}
+		return true
+	}).WithTimeout(testTimeout).Should(BeFalse(), "The PipelineRollout should have been deleted but it was found.")
+
+	document("Verifying Pipeline deletion")
+	Eventually(func() bool {
+		list, err := dynamicClient.Resource(getGVRForPipeline()).Namespace(Namespace).List(ctx, metav1.ListOptions{})
+		if err != nil {
+			return false
+		}
+		if len(list.Items) == 0 {
+			return true
+		}
+		return false
+	}).WithTimeout(testTimeout).Should(BeTrue(), "The Pipeline should have been deleted but it was found.")
+}
+
+// update PipelineRollout and verify correct process
+func updatePipelineRollout(name string, newSpec numaflowv1.PipelineSpec, expectedPhase numaflowv1.PipelinePhase, dataLoss bool) {
```

Comment:
Can we either name `expectedPhase` -> `expectedFinalPhase` or clarify in a comment what it represents? I don't want someone to get confused thinking that if it's a PPND strategy, that represents the temporary "Pausing" state.